[Bug]: Can not create nfs export policy rule - "entry doesn't exist", Target:"svm.uuid"

### Terraform Core Version

1.9.5

### ONTAP Provider Version

1.1.4

### Affected Resource(s)

netapp-ontap_protocols_nfs_export_policy_rule_resource

### Expected Behavior

I am able to create/modify an nfs export policy through terraform provider

### Actual Behavior

 error on POST protocols/nfs/export-policies/0/rules: REST reported error restclient.RestError{Code:"4", Message:"entry doesn't exist", Target:"svm.uuid"},
│ statusCode: 404, statusCode 404

### Relevant Error/Panic Output Snippet

```shell
netapp-ontap_protocols_nfs_export_policy_rule_resource.example: Creating...
╷
│ Error: error creating export policy rule
│ 
│   with netapp-ontap_protocols_nfs_export_policy_rule_resource.example,
│   on export_policy.tf line 1, in resource "netapp-ontap_protocols_nfs_export_policy_rule_resource" "example":
│    1: resource "netapp-ontap_protocols_nfs_export_policy_rule_resource" "example" {
│ 
│ error on POST protocols/nfs/export-policies/0/rules: REST reported error restclient.RestError{Code:"4", Message:"entry doesn't exist", Target:"svm.uuid"},
│ statusCode: 404, statusCode 404
╵
```


### Terraform Configuration Files

terraform {
  required_providers {
    netapp-ontap = {
      source = "NetApp/netapp-ontap"
      version = "1.1.4"
    }
  }
}

provider "netapp-ontap" {
  # A connection profile defines how to interface with an ONTAP cluster or svm.
  # At least one is required.
  connection_profiles = [
    {
      name = "stl"
      hostname = var.stl_hostname
      username = var.stl_username
      password = var.stl_password
      validate_certs = var.validate_certs
    },
    {
      name = "aws"
      hostname = var.aws_hostname
      username = var.aws_username
      password = var.aws_password
      validate_certs = var.validate_certs
    },
  ]
}
variable "stl_svmname" {
    type = string
}

variable "stl_hostname" {
    type = string
}

variable "stl_username" {
    type = string
  
}

variable "stl_password" {
    type = string
  
}

variable "aws_svmname" {
    type = string
  
}

variable "aws_hostname" {
    type = string
  
}

variable "aws_username" {
    type = string
  
}

variable "aws_password" {
    type = string
  
}

variable "aws_svmuuid" {
    type = string
  
}

variable "validate_certs" {
    type = bool
}
resource "netapp-ontap_protocols_nfs_export_policy_rule_resource" "example" {
  cx_profile_name = "aws"
  svm_name = "vs-aws-appsdev"
  export_policy_name = "open"
  clients_match = ["0.0.0.0/0"]
  ro_rule =  ["any"]
  rw_rule =  ["any"]
}

### Steps to Reproduce

just run terraform apply

### Debug Output

> terraform apply --auto-approve
2024-09-12T07:26:26.227-0500 [INFO]  Terraform version: 1.9.5
2024-09-12T07:26:26.227-0500 [DEBUG] using github.com/hashicorp/go-tfe v1.58.0
2024-09-12T07:26:26.227-0500 [DEBUG] using github.com/hashicorp/hcl/v2 v2.20.0
2024-09-12T07:26:26.227-0500 [DEBUG] using github.com/hashicorp/terraform-svchost v0.1.1
2024-09-12T07:26:26.227-0500 [DEBUG] using github.com/zclconf/go-cty v1.14.4
2024-09-12T07:26:26.227-0500 [INFO]  Go runtime version: go1.22.5
2024-09-12T07:26:26.227-0500 [INFO]  CLI args: []string{"terraform", "apply", "--auto-approve"}
2024-09-12T07:26:26.227-0500 [DEBUG] Attempting to open CLI config file: /Users/matt/.terraformrc
2024-09-12T07:26:26.227-0500 [DEBUG] File doesn't exist, but doesn't need to. Ignoring.
2024-09-12T07:26:26.228-0500 [DEBUG] ignoring non-existing provider search directory terraform.d/plugins
2024-09-12T07:26:26.228-0500 [DEBUG] ignoring non-existing provider search directory /Users/matt/.terraform.d/plugins
2024-09-12T07:26:26.228-0500 [DEBUG] ignoring non-existing provider search directory /Users/matt/Library/Application Support/io.terraform/plugins
2024-09-12T07:26:26.228-0500 [DEBUG] ignoring non-existing provider search directory /Library/Application Support/io.terraform/plugins
2024-09-12T07:26:26.229-0500 [INFO]  CLI command args: []string{"apply", "--auto-approve"}
2024-09-12T07:26:26.247-0500 [DEBUG] checking for provisioner in "."
2024-09-12T07:26:26.248-0500 [DEBUG] checking for provisioner in "/usr/local/bin"
2024-09-12T07:26:26.251-0500 [INFO]  backend/local: starting Apply operation
2024-09-12T07:26:26.258-0500 [DEBUG] created provider logger: level=debug
2024-09-12T07:26:26.258-0500 [INFO]  provider: configuring client automatic mTLS
2024-09-12T07:26:26.268-0500 [DEBUG] provider: starting plugin: path=.terraform/providers/registry.terraform.io/netapp/netapp-ontap/1.1.4/darwin_arm64/terraform-provider-netapp-ontap_v1.1.4 args=[".terraform/providers/registry.terraform.io/netapp/netapp-ontap/1.1.4/darwin_arm64/terraform-provider-netapp-ontap_v1.1.4"]
2024-09-12T07:26:26.271-0500 [DEBUG] provider: plugin started: path=.terraform/providers/registry.terraform.io/netapp/netapp-ontap/1.1.4/darwin_arm64/terraform-provider-netapp-ontap_v1.1.4 pid=32897
2024-09-12T07:26:26.271-0500 [DEBUG] provider: waiting for RPC address: plugin=.terraform/providers/registry.terraform.io/netapp/netapp-ontap/1.1.4/darwin_arm64/terraform-provider-netapp-ontap_v1.1.4
2024-09-12T07:26:26.281-0500 [INFO]  provider.terraform-provider-netapp-ontap_v1.1.4: configuring server automatic mTLS: timestamp=2024-09-12T07:26:26.281-0500
2024-09-12T07:26:26.287-0500 [DEBUG] provider: using plugin: version=6
2024-09-12T07:26:26.287-0500 [DEBUG] provider.terraform-provider-netapp-ontap_v1.1.4: plugin address: address=/var/folders/3z/lm73twwj4hb165sscr_27ln80000gn/T/plugin3181896625 network=unix timestamp=2024-09-12T07:26:26.287-0500
2024-09-12T07:26:26.307-0500 [DEBUG] No provider meta schema returned
2024-09-12T07:26:26.309-0500 [DEBUG] provider.stdio: received EOF, stopping recv loop: err="rpc error: code = Unavailable desc = error reading from server: EOF"
2024-09-12T07:26:26.309-0500 [INFO]  provider: plugin process exited: plugin=.terraform/providers/registry.terraform.io/netapp/netapp-ontap/1.1.4/darwin_arm64/terraform-provider-netapp-ontap_v1.1.4 id=32897
2024-09-12T07:26:26.309-0500 [DEBUG] provider: plugin exited
2024-09-12T07:26:26.310-0500 [DEBUG] Building and walking validate graph
2024-09-12T07:26:26.312-0500 [DEBUG] ProviderTransformer: "netapp-ontap_protocols_nfs_export_policy_rule_resource.example" (*terraform.NodeValidatableResource) needs provider["registry.terraform.io/netapp/netapp-ontap"]
2024-09-12T07:26:26.312-0500 [DEBUG] ReferenceTransformer: "provider[\"registry.terraform.io/netapp/netapp-ontap\"]" references: [var.stl_hostname var.stl_username var.stl_password var.validate_certs var.aws_hostname var.aws_username var.aws_password var.validate_certs]
2024-09-12T07:26:26.312-0500 [DEBUG] ReferenceTransformer: "var.stl_svmname" references: []
2024-09-12T07:26:26.312-0500 [DEBUG] ReferenceTransformer: "var.stl_password" references: []
2024-09-12T07:26:26.312-0500 [DEBUG] ReferenceTransformer: "var.validate_certs" references: []
2024-09-12T07:26:26.312-0500 [DEBUG] ReferenceTransformer: "var.aws_password" references: []
2024-09-12T07:26:26.312-0500 [DEBUG] ReferenceTransformer: "var.stl_hostname" references: []
2024-09-12T07:26:26.312-0500 [DEBUG] ReferenceTransformer: "var.aws_hostname" references: []
2024-09-12T07:26:26.312-0500 [DEBUG] ReferenceTransformer: "var.aws_username" references: []
2024-09-12T07:26:26.312-0500 [DEBUG] ReferenceTransformer: "netapp-ontap_protocols_nfs_export_policy_rule_resource.example" references: []
2024-09-12T07:26:26.312-0500 [DEBUG] ReferenceTransformer: "var.stl_username" references: []
2024-09-12T07:26:26.312-0500 [DEBUG] ReferenceTransformer: "var.aws_svmname" references: []
2024-09-12T07:26:26.312-0500 [DEBUG] ReferenceTransformer: "var.aws_svmuuid" references: []
2024-09-12T07:26:26.312-0500 [DEBUG] Starting graph walk: walkValidate
2024-09-12T07:26:26.316-0500 [DEBUG] created provider logger: level=debug
2024-09-12T07:26:26.316-0500 [INFO]  provider: configuring client automatic mTLS
2024-09-12T07:26:26.318-0500 [DEBUG] provider: starting plugin: path=.terraform/providers/registry.terraform.io/netapp/netapp-ontap/1.1.4/darwin_arm64/terraform-provider-netapp-ontap_v1.1.4 args=[".terraform/providers/registry.terraform.io/netapp/netapp-ontap/1.1.4/darwin_arm64/terraform-provider-netapp-ontap_v1.1.4"]
2024-09-12T07:26:26.320-0500 [DEBUG] provider: plugin started: path=.terraform/providers/registry.terraform.io/netapp/netapp-ontap/1.1.4/darwin_arm64/terraform-provider-netapp-ontap_v1.1.4 pid=32899
2024-09-12T07:26:26.320-0500 [DEBUG] provider: waiting for RPC address: plugin=.terraform/providers/registry.terraform.io/netapp/netapp-ontap/1.1.4/darwin_arm64/terraform-provider-netapp-ontap_v1.1.4
2024-09-12T07:26:26.325-0500 [INFO]  provider.terraform-provider-netapp-ontap_v1.1.4: configuring server automatic mTLS: timestamp=2024-09-12T07:26:26.325-0500
2024-09-12T07:26:26.331-0500 [DEBUG] provider: using plugin: version=6
2024-09-12T07:26:26.331-0500 [DEBUG] provider.terraform-provider-netapp-ontap_v1.1.4: plugin address: address=/var/folders/3z/lm73twwj4hb165sscr_27ln80000gn/T/plugin3343118018 network=unix timestamp=2024-09-12T07:26:26.331-0500
2024-09-12T07:26:26.337-0500 [DEBUG] skipping FixUpBlockAttrs
2024-09-12T07:26:26.341-0500 [DEBUG] provider.stdio: received EOF, stopping recv loop: err="rpc error: code = Unavailable desc = error reading from server: EOF"
2024-09-12T07:26:26.341-0500 [INFO]  provider: plugin process exited: plugin=.terraform/providers/registry.terraform.io/netapp/netapp-ontap/1.1.4/darwin_arm64/terraform-provider-netapp-ontap_v1.1.4 id=32899
2024-09-12T07:26:26.341-0500 [DEBUG] provider: plugin exited
2024-09-12T07:26:26.342-0500 [INFO]  backend/local: apply calling Plan
2024-09-12T07:26:26.342-0500 [DEBUG] Building and walking plan graph for NormalMode
2024-09-12T07:26:26.342-0500 [DEBUG] ProviderTransformer: "netapp-ontap_protocols_nfs_export_policy_rule_resource.example (expand)" (*terraform.nodeExpandPlannableResource) needs provider["registry.terraform.io/netapp/netapp-ontap"]
2024-09-12T07:26:26.343-0500 [DEBUG] ReferenceTransformer: "var.validate_certs" references: []
2024-09-12T07:26:26.343-0500 [DEBUG] ReferenceTransformer: "var.stl_svmname" references: []
2024-09-12T07:26:26.343-0500 [DEBUG] ReferenceTransformer: "var.stl_username" references: []
2024-09-12T07:26:26.343-0500 [DEBUG] ReferenceTransformer: "var.stl_password" references: []
2024-09-12T07:26:26.343-0500 [DEBUG] ReferenceTransformer: "var.aws_svmname" references: []
2024-09-12T07:26:26.343-0500 [DEBUG] ReferenceTransformer: "var.aws_svmuuid" references: []
2024-09-12T07:26:26.343-0500 [DEBUG] ReferenceTransformer: "var.aws_hostname" references: []
2024-09-12T07:26:26.343-0500 [DEBUG] ReferenceTransformer: "var.aws_username" references: []
2024-09-12T07:26:26.343-0500 [DEBUG] ReferenceTransformer: "var.aws_password" references: []
2024-09-12T07:26:26.343-0500 [DEBUG] ReferenceTransformer: "provider[\"registry.terraform.io/netapp/netapp-ontap\"]" references: [var.stl_hostname var.stl_username var.stl_password var.validate_certs var.aws_hostname var.aws_username var.aws_password var.validate_certs]
2024-09-12T07:26:26.343-0500 [DEBUG] ReferenceTransformer: "netapp-ontap_protocols_nfs_export_policy_rule_resource.example (expand)" references: []
2024-09-12T07:26:26.343-0500 [DEBUG] ReferenceTransformer: "var.stl_hostname" references: []
2024-09-12T07:26:26.343-0500 [DEBUG] Starting graph walk: walkPlan
2024-09-12T07:26:26.343-0500 [DEBUG] created provider logger: level=debug
2024-09-12T07:26:26.343-0500 [INFO]  provider: configuring client automatic mTLS
2024-09-12T07:26:26.346-0500 [DEBUG] provider: starting plugin: path=.terraform/providers/registry.terraform.io/netapp/netapp-ontap/1.1.4/darwin_arm64/terraform-provider-netapp-ontap_v1.1.4 args=[".terraform/providers/registry.terraform.io/netapp/netapp-ontap/1.1.4/darwin_arm64/terraform-provider-netapp-ontap_v1.1.4"]
2024-09-12T07:26:26.348-0500 [DEBUG] provider: plugin started: path=.terraform/providers/registry.terraform.io/netapp/netapp-ontap/1.1.4/darwin_arm64/terraform-provider-netapp-ontap_v1.1.4 pid=32900
2024-09-12T07:26:26.348-0500 [DEBUG] provider: waiting for RPC address: plugin=.terraform/providers/registry.terraform.io/netapp/netapp-ontap/1.1.4/darwin_arm64/terraform-provider-netapp-ontap_v1.1.4
2024-09-12T07:26:26.353-0500 [INFO]  provider.terraform-provider-netapp-ontap_v1.1.4: configuring server automatic mTLS: timestamp=2024-09-12T07:26:26.352-0500
2024-09-12T07:26:26.359-0500 [DEBUG] provider: using plugin: version=6
2024-09-12T07:26:26.359-0500 [DEBUG] provider.terraform-provider-netapp-ontap_v1.1.4: plugin address: address=/var/folders/3z/lm73twwj4hb165sscr_27ln80000gn/T/plugin1193086403 network=unix timestamp=2024-09-12T07:26:26.359-0500
2024-09-12T07:26:26.364-0500 [DEBUG] skipping FixUpBlockAttrs
2024-09-12T07:26:26.366-0500 [DEBUG] Resource instance state not found for node "netapp-ontap_protocols_nfs_export_policy_rule_resource.example", instance netapp-ontap_protocols_nfs_export_policy_rule_resource.example
2024-09-12T07:26:26.366-0500 [DEBUG] ReferenceTransformer: "netapp-ontap_protocols_nfs_export_policy_rule_resource.example" references: []
2024-09-12T07:26:26.366-0500 [DEBUG] refresh: netapp-ontap_protocols_nfs_export_policy_rule_resource.example: no state, so not refreshing
2024-09-12T07:26:26.369-0500 [DEBUG] provider.terraform-provider-netapp-ontap_v1.1.4: Marking Computed attributes with null configuration values as unknown (known after apply) in the plan to prevent potential Terraform errors: @module=sdk.framework tf_provider_addr=registry.terraform.io/netapp/netapp-ontap tf_resource_type=netapp-ontap_protocols_nfs_export_policy_rule_resource tf_rpc=PlanResourceChange @caller=github.com/hashicorp/terraform-plugin-framework@v1.7.0/internal/fwserver/server_planresourcechange.go:195 tf_req_id=b01581c8-fc45-1a01-9ceb-69251fc41b81 timestamp=2024-09-12T07:26:26.369-0500
2024-09-12T07:26:26.369-0500 [DEBUG] provider.terraform-provider-netapp-ontap_v1.1.4: marking computed attribute that is null in the config as unknown: tf_provider_addr=registry.terraform.io/netapp/netapp-ontap tf_req_id=b01581c8-fc45-1a01-9ceb-69251fc41b81 tf_rpc=PlanResourceChange @caller=github.com/hashicorp/terraform-plugin-framework@v1.7.0/internal/fwserver/server_planresourcechange.go:409 @module=sdk.framework tf_attribute_path="AttributeName(\"index\")" tf_resource_type=netapp-ontap_protocols_nfs_export_policy_rule_resource timestamp=2024-09-12T07:26:26.369-0500
2024-09-12T07:26:26.369-0500 [DEBUG] provider.terraform-provider-netapp-ontap_v1.1.4: marking computed attribute that is null in the config as unknown: @caller=github.com/hashicorp/terraform-plugin-framework@v1.7.0/internal/fwserver/server_planresourcechange.go:409 tf_attribute_path="AttributeName(\"id\")" tf_req_id=b01581c8-fc45-1a01-9ceb-69251fc41b81 tf_resource_type=netapp-ontap_protocols_nfs_export_policy_rule_resource tf_rpc=PlanResourceChange @module=sdk.framework tf_provider_addr=registry.terraform.io/netapp/netapp-ontap timestamp=2024-09-12T07:26:26.369-0500
2024-09-12T07:26:26.369-0500 [DEBUG] provider.terraform-provider-netapp-ontap_v1.1.4: marking computed attribute that is null in the config as unknown: @module=sdk.framework tf_req_id=b01581c8-fc45-1a01-9ceb-69251fc41b81 tf_resource_type=netapp-ontap_protocols_nfs_export_policy_rule_resource tf_rpc=PlanResourceChange tf_provider_addr=registry.terraform.io/netapp/netapp-ontap @caller=github.com/hashicorp/terraform-plugin-framework@v1.7.0/internal/fwserver/server_planresourcechange.go:409 tf_attribute_path="AttributeName(\"export_policy_id\")" timestamp=2024-09-12T07:26:26.369-0500
2024-09-12T07:26:26.371-0500 [DEBUG] provider.stdio: received EOF, stopping recv loop: err="rpc error: code = Unavailable desc = error reading from server: EOF"
2024-09-12T07:26:26.372-0500 [INFO]  provider: plugin process exited: plugin=.terraform/providers/registry.terraform.io/netapp/netapp-ontap/1.1.4/darwin_arm64/terraform-provider-netapp-ontap_v1.1.4 id=32900
2024-09-12T07:26:26.372-0500 [DEBUG] provider: plugin exited
2024-09-12T07:26:26.372-0500 [DEBUG] building apply graph to check for errors
2024-09-12T07:26:26.372-0500 [DEBUG] Resource state not found for node "netapp-ontap_protocols_nfs_export_policy_rule_resource.example", instance netapp-ontap_protocols_nfs_export_policy_rule_resource.example
2024-09-12T07:26:26.372-0500 [DEBUG] ProviderTransformer: "netapp-ontap_protocols_nfs_export_policy_rule_resource.example" (*terraform.NodeApplyableResourceInstance) needs provider["registry.terraform.io/netapp/netapp-ontap"]
2024-09-12T07:26:26.372-0500 [DEBUG] ProviderTransformer: "netapp-ontap_protocols_nfs_export_policy_rule_resource.example (expand)" (*terraform.nodeExpandApplyableResource) needs provider["registry.terraform.io/netapp/netapp-ontap"]
2024-09-12T07:26:26.373-0500 [DEBUG] ReferenceTransformer: "var.aws_svmname" references: []
2024-09-12T07:26:26.373-0500 [DEBUG] ReferenceTransformer: "var.aws_svmuuid" references: []
2024-09-12T07:26:26.373-0500 [DEBUG] ReferenceTransformer: "var.validate_certs" references: []
2024-09-12T07:26:26.373-0500 [DEBUG] ReferenceTransformer: "var.stl_username" references: []
2024-09-12T07:26:26.373-0500 [DEBUG] ReferenceTransformer: "var.stl_password" references: []
2024-09-12T07:26:26.373-0500 [DEBUG] ReferenceTransformer: "netapp-ontap_protocols_nfs_export_policy_rule_resource.example" references: []
2024-09-12T07:26:26.373-0500 [DEBUG] ReferenceTransformer: "netapp-ontap_protocols_nfs_export_policy_rule_resource.example (expand)" references: []
2024-09-12T07:26:26.373-0500 [DEBUG] ReferenceTransformer: "var.aws_hostname" references: []
2024-09-12T07:26:26.373-0500 [DEBUG] ReferenceTransformer: "var.aws_username" references: []
2024-09-12T07:26:26.373-0500 [DEBUG] ReferenceTransformer: "var.stl_svmname" references: []
2024-09-12T07:26:26.373-0500 [DEBUG] ReferenceTransformer: "provider[\"registry.terraform.io/netapp/netapp-ontap\"]" references: [var.stl_hostname var.stl_username var.stl_password var.validate_certs var.aws_hostname var.aws_username var.aws_password var.validate_certs]
2024-09-12T07:26:26.373-0500 [DEBUG] ReferenceTransformer: "var.aws_password" references: []
2024-09-12T07:26:26.373-0500 [DEBUG] ReferenceTransformer: "var.stl_hostname" references: []

Terraform used the selected providers to generate the following execution plan. Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  # netapp-ontap_protocols_nfs_export_policy_rule_resource.example will be created
  + resource "netapp-ontap_protocols_nfs_export_policy_rule_resource" "example" {
      + allow_device_creation = true
      + allow_suid            = true
      + anonymous_user        = "65534"
      + chown_mode            = "restricted"
      + clients_match         = [
          + "0.0.0.0/0",
        ]
      + cx_profile_name       = "aws"
      + export_policy_id      = (known after apply)
      + export_policy_name    = "open"
      + id                    = (known after apply)
      + index                 = (known after apply)
      + ntfs_unix_security    = "fail"
      + protocols             = [
          + "any",
        ]
      + ro_rule               = [
          + "any",
        ]
      + rw_rule               = [
          + "any",
        ]
      + superuser             = [
          + "any",
        ]
      + svm_name              = "vs-aws-appsdev"
    }

Plan: 1 to add, 0 to change, 0 to destroy.
2024-09-12T07:26:26.374-0500 [INFO]  backend/local: apply calling Apply
2024-09-12T07:26:26.374-0500 [DEBUG] Building and walking apply graph for NormalMode plan
2024-09-12T07:26:26.374-0500 [DEBUG] Resource state not found for node "netapp-ontap_protocols_nfs_export_policy_rule_resource.example", instance netapp-ontap_protocols_nfs_export_policy_rule_resource.example
2024-09-12T07:26:26.375-0500 [DEBUG] ProviderTransformer: "netapp-ontap_protocols_nfs_export_policy_rule_resource.example" (*terraform.NodeApplyableResourceInstance) needs provider["registry.terraform.io/netapp/netapp-ontap"]
2024-09-12T07:26:26.375-0500 [DEBUG] ProviderTransformer: "netapp-ontap_protocols_nfs_export_policy_rule_resource.example (expand)" (*terraform.nodeExpandApplyableResource) needs provider["registry.terraform.io/netapp/netapp-ontap"]
2024-09-12T07:26:26.375-0500 [DEBUG] ReferenceTransformer: "var.stl_username" references: []
2024-09-12T07:26:26.375-0500 [DEBUG] ReferenceTransformer: "var.stl_password" references: []
2024-09-12T07:26:26.375-0500 [DEBUG] ReferenceTransformer: "var.aws_svmuuid" references: []
2024-09-12T07:26:26.375-0500 [DEBUG] ReferenceTransformer: "var.aws_username" references: []
2024-09-12T07:26:26.375-0500 [DEBUG] ReferenceTransformer: "var.aws_password" references: []
2024-09-12T07:26:26.375-0500 [DEBUG] ReferenceTransformer: "netapp-ontap_protocols_nfs_export_policy_rule_resource.example (expand)" references: []
2024-09-12T07:26:26.375-0500 [DEBUG] ReferenceTransformer: "var.stl_svmname" references: []
2024-09-12T07:26:26.375-0500 [DEBUG] ReferenceTransformer: "var.aws_svmname" references: []
2024-09-12T07:26:26.375-0500 [DEBUG] ReferenceTransformer: "var.validate_certs" references: []
2024-09-12T07:26:26.375-0500 [DEBUG] ReferenceTransformer: "var.stl_hostname" references: []
2024-09-12T07:26:26.375-0500 [DEBUG] ReferenceTransformer: "var.aws_hostname" references: []
2024-09-12T07:26:26.375-0500 [DEBUG] ReferenceTransformer: "netapp-ontap_protocols_nfs_export_policy_rule_resource.example" references: []
2024-09-12T07:26:26.375-0500 [DEBUG] ReferenceTransformer: "provider[\"registry.terraform.io/netapp/netapp-ontap\"]" references: [var.stl_hostname var.stl_username var.stl_password var.validate_certs var.aws_hostname var.aws_username var.aws_password var.validate_certs]
2024-09-12T07:26:26.375-0500 [DEBUG] Starting graph walk: walkApply
2024-09-12T07:26:26.375-0500 [DEBUG] created provider logger: level=debug
2024-09-12T07:26:26.375-0500 [INFO]  provider: configuring client automatic mTLS
2024-09-12T07:26:26.378-0500 [DEBUG] provider: starting plugin: path=.terraform/providers/registry.terraform.io/netapp/netapp-ontap/1.1.4/darwin_arm64/terraform-provider-netapp-ontap_v1.1.4 args=[".terraform/providers/registry.terraform.io/netapp/netapp-ontap/1.1.4/darwin_arm64/terraform-provider-netapp-ontap_v1.1.4"]
2024-09-12T07:26:26.379-0500 [DEBUG] provider: plugin started: path=.terraform/providers/registry.terraform.io/netapp/netapp-ontap/1.1.4/darwin_arm64/terraform-provider-netapp-ontap_v1.1.4 pid=32901
2024-09-12T07:26:26.379-0500 [DEBUG] provider: waiting for RPC address: plugin=.terraform/providers/registry.terraform.io/netapp/netapp-ontap/1.1.4/darwin_arm64/terraform-provider-netapp-ontap_v1.1.4
2024-09-12T07:26:26.384-0500 [INFO]  provider.terraform-provider-netapp-ontap_v1.1.4: configuring server automatic mTLS: timestamp=2024-09-12T07:26:26.384-0500
2024-09-12T07:26:26.390-0500 [DEBUG] provider: using plugin: version=6
2024-09-12T07:26:26.390-0500 [DEBUG] provider.terraform-provider-netapp-ontap_v1.1.4: plugin address: address=/var/folders/3z/lm73twwj4hb165sscr_27ln80000gn/T/plugin3932036486 network=unix timestamp=2024-09-12T07:26:26.390-0500
2024-09-12T07:26:26.395-0500 [DEBUG] skipping FixUpBlockAttrs
2024-09-12T07:26:26.399-0500 [DEBUG] provider.terraform-provider-netapp-ontap_v1.1.4: Marking Computed attributes with null configuration values as unknown (known after apply) in the plan to prevent potential Terraform errors: tf_rpc=PlanResourceChange tf_resource_type=netapp-ontap_protocols_nfs_export_policy_rule_resource @caller=github.com/hashicorp/terraform-plugin-framework@v1.7.0/internal/fwserver/server_planresourcechange.go:195 @module=sdk.framework tf_provider_addr=registry.terraform.io/netapp/netapp-ontap tf_req_id=805b1efb-2e9a-6f60-bacf-2eb419eceb99 timestamp=2024-09-12T07:26:26.399-0500
2024-09-12T07:26:26.399-0500 [DEBUG] provider.terraform-provider-netapp-ontap_v1.1.4: marking computed attribute that is null in the config as unknown: @caller=github.com/hashicorp/terraform-plugin-framework@v1.7.0/internal/fwserver/server_planresourcechange.go:409 tf_req_id=805b1efb-2e9a-6f60-bacf-2eb419eceb99 tf_resource_type=netapp-ontap_protocols_nfs_export_policy_rule_resource @module=sdk.framework tf_attribute_path="AttributeName(\"export_policy_id\")" tf_provider_addr=registry.terraform.io/netapp/netapp-ontap tf_rpc=PlanResourceChange timestamp=2024-09-12T07:26:26.399-0500
2024-09-12T07:26:26.399-0500 [DEBUG] provider.terraform-provider-netapp-ontap_v1.1.4: marking computed attribute that is null in the config as unknown: @module=sdk.framework tf_attribute_path="AttributeName(\"id\")" tf_provider_addr=registry.terraform.io/netapp/netapp-ontap tf_req_id=805b1efb-2e9a-6f60-bacf-2eb419eceb99 @caller=github.com/hashicorp/terraform-plugin-framework@v1.7.0/internal/fwserver/server_planresourcechange.go:409 tf_resource_type=netapp-ontap_protocols_nfs_export_policy_rule_resource tf_rpc=PlanResourceChange timestamp=2024-09-12T07:26:26.399-0500
2024-09-12T07:26:26.399-0500 [DEBUG] provider.terraform-provider-netapp-ontap_v1.1.4: marking computed attribute that is null in the config as unknown: tf_attribute_path="AttributeName(\"index\")" tf_req_id=805b1efb-2e9a-6f60-bacf-2eb419eceb99 tf_resource_type=netapp-ontap_protocols_nfs_export_policy_rule_resource tf_rpc=PlanResourceChange @caller=github.com/hashicorp/terraform-plugin-framework@v1.7.0/internal/fwserver/server_planresourcechange.go:409 @module=sdk.framework tf_provider_addr=registry.terraform.io/netapp/netapp-ontap timestamp=2024-09-12T07:26:26.399-0500
netapp-ontap_protocols_nfs_export_policy_rule_resource.example: Creating...
2024-09-12T07:26:26.401-0500 [INFO]  Starting apply for netapp-ontap_protocols_nfs_export_policy_rule_resource.example
2024-09-12T07:26:26.401-0500 [DEBUG] netapp-ontap_protocols_nfs_export_policy_rule_resource.example: applying the planned Create change
2024-09-12T07:26:26.402-0500 [DEBUG] provider.terraform-provider-netapp-ontap_v1.1.4: Version string is: "TerrafromONTAP/protocols_nfs_export_policy_rule_resource/1.1.4": tf_provider_addr=registry.terraform.io/netapp/netapp-ontap tf_req_id=ebfc764e-1d01-c4da-d876-6b3e178d7888 tf_rpc=ApplyResourceChange tf_resource_type=netapp-ontap_protocols_nfs_export_policy_rule_resource @caller=github.com/netapp/terraform-provider-netapp-ontap/internal/provider/config.go:67 @module=netapp_ontap timestamp=2024-09-12T07:26:26.402-0500
2024-09-12T07:26:26.402-0500 [DEBUG] provider.terraform-provider-netapp-ontap_v1.1.4: sending: GET https://172.26.94.140/api/protocols/nfs/export-policies?fields=name&name=open&svm.name=vs-aws-appsdev: @caller=github.com/netapp/terraform-provider-netapp-ontap/internal/restclient/httpclient/http_client.go:45 body=<nil> tf_req_id=ebfc764e-1d01-c4da-d876-6b3e178d7888 tf_resource_type=netapp-ontap_protocols_nfs_export_policy_rule_resource @module=netapp_ontap tf_provider_addr=registry.terraform.io/netapp/netapp-ontap tf_rpc=ApplyResourceChange timestamp=2024-09-12T07:26:26.402-0500
2024-09-12T07:26:26.486-0500 [DEBUG] provider.terraform-provider-netapp-ontap_v1.1.4: received: GET https://172.26.94.140/api/protocols/nfs/export-policies?fields=name&name=open&svm.name=vs-aws-appsdev 200: tf_req_id=ebfc764e-1d01-c4da-d876-6b3e178d7888 tf_resource_type=netapp-ontap_protocols_nfs_export_policy_rule_resource @module=netapp_ontap tf_rpc=ApplyResourceChange @caller=github.com/netapp/terraform-provider-netapp-ontap/internal/restclient/httpclient/http_client.go:67
  res=
  | {
  |   "records": [
  |   ],
  |   "num_records": 0,
  |   "_links": {
  |     "self": {
  |       "href": "/api/protocols/nfs/export-policies?fields=name&name=open&svm.name=vs-aws-appsdev"
  |     }
  |   }
  | }
   tf_provider_addr=registry.terraform.io/netapp/netapp-ontap timestamp=2024-09-12T07:26:26.486-0500
2024-09-12T07:26:26.486-0500 [DEBUG] provider.terraform-provider-netapp-ontap_v1.1.4: dataMap map[string]interface {}{"_links":map[string]interface {}{"self":map[string]interface {}{"href":"/api/protocols/nfs/export-policies?fields=name&name=open&svm.name=vs-aws-appsdev"}}, "num_records":0, "records":[]interface {}{}}: tf_req_id=ebfc764e-1d01-c4da-d876-6b3e178d7888 @module=netapp_ontap tf_provider_addr=registry.terraform.io/netapp/netapp-ontap @caller=github.com/netapp/terraform-provider-netapp-ontap/internal/restclient/rest_response.go:56 tf_resource_type=netapp-ontap_protocols_nfs_export_policy_rule_resource tf_rpc=ApplyResourceChange timestamp=2024-09-12T07:26:26.486-0500
2024-09-12T07:26:26.486-0500 [DEBUG] provider.terraform-provider-netapp-ontap_v1.1.4: rawResponse restclient.restStagedResponse{NumRecords:0, Records:[]map[string]interface {}{}, Error:restclient.RestError{Code:"", Message:"", Target:""}, Job:map[string]interface {}(nil), Jobs:[]map[string]interface {}(nil), Other:map[string]interface {}{"_links":map[string]interface {}{"self":map[string]interface {}{"href":"/api/protocols/nfs/export-policies?fields=name&name=open&svm.name=vs-aws-appsdev"}}}}, metadata mapstructure.Metadata{Keys:[]string{"num_records", "Records", "[<interface {} Value>]", "[<interface {} Value>]"}, Unused:[]string{}, Unset:[]string{"Error", "Job", "Jobs"}}: @caller=github.com/netapp/terraform-provider-netapp-ontap/internal/restclient/rest_response.go:77 tf_provider_addr=registry.terraform.io/netapp/netapp-ontap tf_req_id=ebfc764e-1d01-c4da-d876-6b3e178d7888 @module=netapp_ontap tf_resource_type=netapp-ontap_protocols_nfs_export_policy_rule_resource tf_rpc=ApplyResourceChange timestamp=2024-09-12T07:26:26.486-0500
2024-09-12T07:26:26.486-0500 [DEBUG] provider.terraform-provider-netapp-ontap_v1.1.4: finalResponse restclient.RestResponse{NumRecords:0, Records:[]map[string]interface {}{}, RestError:restclient.RestError{Code:"", Message:"", Target:""}, StatusCode:200, HTTPError:"", ErrorType:"", Job:map[string]interface {}(nil), Jobs:[]map[string]interface {}(nil)}, metadata mapstructure.Metadata{Keys:[]string{"num_records", "Records", "[<interface {} Value>]", "[<interface {} Value>]", "Error", "num_records", "Records", "error.Code", "error.Message", "error.Target", "error", "Job", "Jobs"}, Unused:[]string{"Other"}, Unset:[]string{"Error", "Job", "Jobs", "StatusCode", "HTTPError", "ErrorType"}}: tf_provider_addr=registry.terraform.io/netapp/netapp-ontap tf_resource_type=netapp-ontap_protocols_nfs_export_policy_rule_resource tf_rpc=ApplyResourceChange @caller=github.com/netapp/terraform-provider-netapp-ontap/internal/restclient/rest_response.go:100 @module=netapp_ontap tf_req_id=ebfc764e-1d01-c4da-d876-6b3e178d7888 timestamp=2024-09-12T07:26:26.486-0500
2024-09-12T07:26:26.486-0500 [DEBUG] provider.terraform-provider-netapp-ontap_v1.1.4: Read export policy source - udata: interfaces.ExportPolicyGetDataModelONTAP{Name:"", Svm:"", ID:0}: tf_provider_addr=registry.terraform.io/netapp/netapp-ontap tf_req_id=ebfc764e-1d01-c4da-d876-6b3e178d7888 tf_rpc=ApplyResourceChange @module=netapp_ontap tf_resource_type=netapp-ontap_protocols_nfs_export_policy_rule_resource @caller=github.com/netapp/terraform-provider-netapp-ontap/internal/interfaces/protocols_nfs_export_policy.go:92 timestamp=2024-09-12T07:26:26.486-0500
2024-09-12T07:26:26.486-0500 [DEBUG] provider.terraform-provider-netapp-ontap_v1.1.4: sending: POST https://172.26.94.140/api/protocols/nfs/export-policies/0/rules?return_records=true&return_timeout=60: @caller=github.com/netapp/terraform-provider-netapp-ontap/internal/restclient/httpclient/http_client.go:45 tf_provider_addr=registry.terraform.io/netapp/netapp-ontap tf_req_id=ebfc764e-1d01-c4da-d876-6b3e178d7888 tf_resource_type=netapp-ontap_protocols_nfs_export_policy_rule_resource tf_rpc=ApplyResourceChange @module=netapp_ontap body="map[allow_device_creation:true allow_suid:true anonymous_user:65534 chown_mode:restricted clients:[map[match:0.0.0.0/0]] ntfs_unix_security:fail protocols:[any] ro_rule:[any] rw_rule:[any] superuser:[any]]" timestamp=2024-09-12T07:26:26.486-0500
2024-09-12T07:26:27.074-0500 [DEBUG] provider.terraform-provider-netapp-ontap_v1.1.4: received: POST https://172.26.94.140/api/protocols/nfs/export-policies/0/rules?return_records=true&return_timeout=60 404: tf_rpc=ApplyResourceChange @caller=github.com/netapp/terraform-provider-netapp-ontap/internal/restclient/httpclient/http_client.go:67 @module=netapp_ontap
  res=
  | {
  |   "error": {
  |     "message": "entry doesn't exist",
  |     "code": "4",
  |     "target": "svm.uuid"
  |   }
  | }
   tf_provider_addr=registry.terraform.io/netapp/netapp-ontap tf_req_id=ebfc764e-1d01-c4da-d876-6b3e178d7888 tf_resource_type=netapp-ontap_protocols_nfs_export_policy_rule_resource timestamp=2024-09-12T07:26:27.073-0500
2024-09-12T07:26:27.074-0500 [DEBUG] provider.terraform-provider-netapp-ontap_v1.1.4: dataMap map[string]interface {}{"error":map[string]interface {}{"code":"4", "message":"entry doesn't exist", "target":"svm.uuid"}}: @module=netapp_ontap tf_provider_addr=registry.terraform.io/netapp/netapp-ontap tf_req_id=ebfc764e-1d01-c4da-d876-6b3e178d7888 tf_resource_type=netapp-ontap_protocols_nfs_export_policy_rule_resource @caller=github.com/netapp/terraform-provider-netapp-ontap/internal/restclient/rest_response.go:56 tf_rpc=ApplyResourceChange timestamp=2024-09-12T07:26:27.074-0500
2024-09-12T07:26:27.074-0500 [DEBUG] provider.terraform-provider-netapp-ontap_v1.1.4: rawResponse restclient.restStagedResponse{NumRecords:0, Records:[]map[string]interface {}(nil), Error:restclient.RestError{Code:"4", Message:"entry doesn't exist", Target:"svm.uuid"}, Job:map[string]interface {}(nil), Jobs:[]map[string]interface {}(nil), Other:map[string]interface {}(nil)}, metadata mapstructure.Metadata{Keys:[]string{"Error.Code", "Error.Message", "Error.Target", "Error"}, Unused:[]string{}, Unset:[]string{"Job", "Jobs", "num_records", "Records"}}: @module=netapp_ontap tf_req_id=ebfc764e-1d01-c4da-d876-6b3e178d7888 @caller=github.com/netapp/terraform-provider-netapp-ontap/internal/restclient/rest_response.go:77 tf_provider_addr=registry.terraform.io/netapp/netapp-ontap tf_resource_type=netapp-ontap_protocols_nfs_export_policy_rule_resource tf_rpc=ApplyResourceChange timestamp=2024-09-12T07:26:27.074-0500
2024-09-12T07:26:27.075-0500 [ERROR] provider.terraform-provider-netapp-ontap_v1.1.4: checkRestError: REST reported error restclient.RestError{Code:"4", Message:"entry doesn't exist", Target:"svm.uuid"}, statusCode: 404, statusCode 404, response: restclient.RestResponse{NumRecords:0, Records:[]map[string]interface {}(nil), RestError:restclient.RestError{Code:"4", Message:"entry doesn't exist", Target:"svm.uuid"}, StatusCode:404, HTTPError:"", ErrorType:"rest_error", Job:map[string]interface {}(nil), Jobs:[]map[string]interface {}(nil)}: tf_req_id=ebfc764e-1d01-c4da-d876-6b3e178d7888 tf_resource_type=netapp-ontap_protocols_nfs_export_policy_rule_resource @module=netapp_ontap tf_provider_addr=registry.terraform.io/netapp/netapp-ontap tf_rpc=ApplyResourceChange @caller=github.com/netapp/terraform-provider-netapp-ontap/internal/restclient/rest_response.go:114 timestamp=2024-09-12T07:26:27.074-0500
2024-09-12T07:26:27.075-0500 [DEBUG] provider.terraform-provider-netapp-ontap_v1.1.4: finalResponse restclient.RestResponse{NumRecords:0, Records:[]map[string]interface {}(nil), RestError:restclient.RestError{Code:"4", Message:"entry doesn't exist", Target:"svm.uuid"}, StatusCode:404, HTTPError:"", ErrorType:"rest_error", Job:map[string]interface {}(nil), Jobs:[]map[string]interface {}(nil)}, metadata mapstructure.Metadata{Keys:[]string{"Error.Code", "Error.Message", "Error.Target", "Error", "Error", "num_records", "Records", "error.Code", "error.Message", "error.Target", "error", "Job", "Jobs"}, Unused:[]string{"Other"}, Unset:[]string{"Job", "Jobs", "num_records", "Records", "StatusCode", "HTTPError", "ErrorType"}}: @module=netapp_ontap tf_resource_type=netapp-ontap_protocols_nfs_export_policy_rule_resource tf_rpc=ApplyResourceChange tf_req_id=ebfc764e-1d01-c4da-d876-6b3e178d7888 @caller=github.com/netapp/terraform-provider-netapp-ontap/internal/restclient/rest_response.go:100 tf_provider_addr=registry.terraform.io/netapp/netapp-ontap timestamp=2024-09-12T07:26:27.074-0500
2024-09-12T07:26:27.075-0500 [DEBUG] provider.terraform-provider-netapp-ontap_v1.1.4: CallCreateMethod request failed 404: tf_req_id=ebfc764e-1d01-c4da-d876-6b3e178d7888 @module=netapp_ontap tf_provider_addr=registry.terraform.io/netapp/netapp-ontap tf_resource_type=netapp-ontap_protocols_nfs_export_policy_rule_resource tf_rpc=ApplyResourceChange @caller=github.com/netapp/terraform-provider-netapp-ontap/internal/restclient/rest_client.go:49 timestamp=2024-09-12T07:26:27.074-0500
2024-09-12T07:26:27.075-0500 [ERROR] provider.terraform-provider-netapp-ontap_v1.1.4: error on POST protocols/nfs/export-policies/0/rules: REST reported error restclient.RestError{Code:"4", Message:"entry doesn't exist", Target:"svm.uuid"}, statusCode: 404, statusCode 404: @caller=github.com/netapp/terraform-provider-netapp-ontap/internal/interfaces/protocols_nfs_export_policy_rule.go:83 @module=netapp_ontap.error_handler timestamp=2024-09-12T07:26:27.074-0500
2024-09-12T07:26:27.075-0500 [ERROR] provider.terraform-provider-netapp-ontap_v1.1.4: Response contains error diagnostic: tf_resource_type=netapp-ontap_protocols_nfs_export_policy_rule_resource @caller=github.com/hashicorp/terraform-plugin-go@v0.22.1/tfprotov6/internal/diag/diagnostics.go:58 @module=sdk.proto diagnostic_detail="error on POST protocols/nfs/export-policies/0/rules: REST reported error restclient.RestError{Code:\"4\", Message:\"entry doesn't exist\", Target:\"svm.uuid\"}, statusCode: 404, statusCode 404" diagnostic_severity=ERROR tf_proto_version=6.4 tf_req_id=ebfc764e-1d01-c4da-d876-6b3e178d7888 diagnostic_summary="error creating export policy rule" tf_provider_addr=registry.terraform.io/netapp/netapp-ontap tf_rpc=ApplyResourceChange timestamp=2024-09-12T07:26:27.074-0500
2024-09-12T07:26:27.084-0500 [DEBUG] State storage *statemgr.Filesystem declined to persist a state snapshot
2024-09-12T07:26:27.084-0500 [ERROR] vertex "netapp-ontap_protocols_nfs_export_policy_rule_resource.example" error: error creating export policy rule
╷
│ Error: error creating export policy rule
│
│   with netapp-ontap_protocols_nfs_export_policy_rule_resource.example,
│   on export_policy.tf line 1, in resource "netapp-ontap_protocols_nfs_export_policy_rule_resource" "example":
│    1: resource "netapp-ontap_protocols_nfs_export_policy_rule_resource" "example" {
│
│ error on POST protocols/nfs/export-policies/0/rules: REST reported error restclient.RestError{Code:"4", Message:"entry doesn't exist", Target:"svm.uuid"}, statusCode: 404, statusCode
│ 404
╵
2024-09-12T07:26:27.097-0500 [DEBUG] provider.stdio: received EOF, stopping recv loop: err="rpc error: code = Unavailable desc = error reading from server: EOF"
2024-09-12T07:26:27.098-0500 [INFO]  provider: plugin process exited: plugin=.terraform/providers/registry.terraform.io/netapp/netapp-ontap/1.1.4/darwin_arm64/terraform-provider-netapp-ontap_v1.1.4 id=32901
2024-09-12T07:26:27.098-0500 [DEBUG] provider: plugin exited

### Panic Output

_No response_

### Important Factoids

we tried against a NetApp CVO instance in AWS running OnTap 9.15.1
We also tried against an on prem FAS running OnTap 9.14.1
we tested as a user inside an SVM and as the admin user of the cluster
all iterations produced the same result

### References

_No response_

### Would you like to implement a fix?

No

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -35,6 +35,7 @@ DOC FIXES:
 
 BUG FIXES:
 * **netapp-ontap_cluster_data_source: fix on nodes to show multiple elements ([#264](https://github.com/NetApp/terraform-provider-netapp-ontap/issues/264))
+* **netapp-ontap_protocols_nfs_export_policy_resource: fix id error during the creation ([[#290](https://github.com/NetApp/terraform-provider-netapp-ontap/issues/290)])
 
 ## 1.1.3 (2024-08-08)
 

--- a/internal/interfaces/protocols_nfs_export_policy.go
+++ b/internal/interfaces/protocols_nfs_export_policy.go
@@ -16,6 +16,12 @@ type ExportpolicyResourceModel struct {
 	ID   int               `mapstructure:"id"`
 }
 
+// ExportpolicyResourceBodyDataModelONTAP describes the resource data model.
+type ExportpolicyResourceBodyDataModelONTAP struct {
+	Name string            `mapstructure:"name"`
+	Svm  SvmDataModelONTAP `mapstructure:"svm"`
+}
+
 // ExportPolicyGetDataModelONTAP describes the GET record data model using go types for mapping.
 type ExportPolicyGetDataModelONTAP struct {
 	Name string `mapstructure:"name"`
@@ -30,7 +36,7 @@ type ExportPolicyGetDataFilterModel struct {
 }
 
 // CreateExportPolicy to create export policy
-func CreateExportPolicy(errorHandler *utils.ErrorHandler, r restclient.RestClient, data ExportpolicyResourceModel) (*ExportPolicyGetDataModelONTAP, error) {
+func CreateExportPolicy(errorHandler *utils.ErrorHandler, r restclient.RestClient, data ExportpolicyResourceBodyDataModelONTAP) (*ExportPolicyGetDataModelONTAP, error) {
 	var body map[string]interface{}
 	if err := mapstructure.Decode(data, &body); err != nil {
 		return nil, errorHandler.MakeAndReportError("error encoding export policy body", fmt.Sprintf("error on encoding export policy body: %s, body: %#v", err, data))
@@ -136,7 +142,7 @@ func DeleteExportPolicy(errorHandler *utils.ErrorHandler, r restclient.RestClien
 }
 
 // UpdateExportPolicy updates export policy
-func UpdateExportPolicy(errorHandler *utils.ErrorHandler, r restclient.RestClient, data ExportpolicyResourceModel, id string) error {
+func UpdateExportPolicy(errorHandler *utils.ErrorHandler, r restclient.RestClient, data ExportpolicyResourceBodyDataModelONTAP, id string) error {
 	var body map[string]interface{}
 	if err := mapstructure.Decode(data, &body); err != nil {
 		return errorHandler.MakeAndReportError("error encoding export policy body", fmt.Sprintf("error on encoding export policy body: %s, body: %#v", err, data))

--- a/internal/interfaces/protocols_nfs_export_policy.go
+++ b/internal/interfaces/protocols_nfs_export_policy.go
@@ -24,9 +24,10 @@ type ExportpolicyResourceBodyDataModelONTAP struct {
 
 // ExportPolicyGetDataModelONTAP describes the GET record data model using go types for mapping.
 type ExportPolicyGetDataModelONTAP struct {
-	Name string `mapstructure:"name"`
-	Svm  string `mapstructure:"svm_name"`
-	ID   int    `mapstructure:"id"`
+	Name    string `mapstructure:"name"`
+	Svm     string `mapstructure:"svm_name"`
+	SvmUUID string `mapstructure:"svm_uuid"`
+	ID      int    `mapstructure:"id"`
 }
 
 // ExportPolicyGetDataFilterModel describes filter model
@@ -36,7 +37,7 @@ type ExportPolicyGetDataFilterModel struct {
 }
 
 // CreateExportPolicy to create export policy
-func CreateExportPolicy(errorHandler *utils.ErrorHandler, r restclient.RestClient, data ExportpolicyResourceBodyDataModelONTAP) (*ExportPolicyGetDataModelONTAP, error) {
+func CreateExportPolicy(errorHandler *utils.ErrorHandler, r restclient.RestClient, data ExportpolicyResourceBodyDataModelONTAP) (*ExportpolicyResourceModel, error) {
 	var body map[string]interface{}
 	if err := mapstructure.Decode(data, &body); err != nil {
 		return nil, errorHandler.MakeAndReportError("error encoding export policy body", fmt.Sprintf("error on encoding export policy body: %s, body: %#v", err, data))
@@ -48,7 +49,7 @@ func CreateExportPolicy(errorHandler *utils.ErrorHandler, r restclient.RestClien
 		return nil, errorHandler.MakeAndReportError("error creating export policy", fmt.Sprintf("error on POST protocols/nfs/export-policies: %s, statusCode %d", err, statusCode))
 	}
 
-	var dataONTAP ExportPolicyGetDataModelONTAP
+	var dataONTAP ExportpolicyResourceModel
 	if err := mapstructure.Decode(response.Records[0], &dataONTAP); err != nil {
 		return nil, errorHandler.MakeAndReportError("error decoding export policies info", fmt.Sprintf("error on decode protocols/nfs/export-policies info: %s, statusCode %d, response %#v", err, statusCode, response))
 	}
@@ -57,7 +58,7 @@ func CreateExportPolicy(errorHandler *utils.ErrorHandler, r restclient.RestClien
 }
 
 // GetExportPolicy to get export policy
-func GetExportPolicy(errorHandler *utils.ErrorHandler, r restclient.RestClient, id string) (*ExportPolicyGetDataModelONTAP, error) {
+func GetExportPolicy(errorHandler *utils.ErrorHandler, r restclient.RestClient, id string) (*ExportpolicyResourceModel, error) {
 	api := "protocols/nfs/export-policies/" + id
 	statusCode, response, err := r.GetNilOrOneRecord(api, nil, nil)
 	if err == nil && response == nil {
@@ -67,7 +68,7 @@ func GetExportPolicy(errorHandler *utils.ErrorHandler, r restclient.RestClient, 
 		return nil, errorHandler.MakeAndReportError("error reading export policy info", fmt.Sprintf("error on GET protocols/nfs/export-policies/%s: %s", id, err))
 	}
 
-	var dataONTAP ExportPolicyGetDataModelONTAP
+	var dataONTAP ExportpolicyResourceModel
 	if err := mapstructure.Decode(response, &dataONTAP); err != nil {
 		return nil, errorHandler.MakeAndReportError("error decoding export policy info", fmt.Sprintf("error on decode protocols/nfs/export-policies/%s: %s, statusCode %d, response %#v", id, err, statusCode, response))
 	}
@@ -76,7 +77,7 @@ func GetExportPolicy(errorHandler *utils.ErrorHandler, r restclient.RestClient, 
 }
 
 // GetNfsExportPolicyByName to get export policy by filter
-func GetNfsExportPolicyByName(errorHandler *utils.ErrorHandler, r restclient.RestClient, filter interface{}) (*ExportPolicyGetDataModelONTAP, error) {
+func GetNfsExportPolicyByName(errorHandler *utils.ErrorHandler, r restclient.RestClient, filter interface{}) (*ExportpolicyResourceModel, error) {
 	query := r.NewQuery()
 	query.Fields([]string{"name"})
 	if filter != nil {
@@ -91,7 +92,7 @@ func GetNfsExportPolicyByName(errorHandler *utils.ErrorHandler, r restclient.Res
 		return nil, errorHandler.MakeAndReportError("error reading export policy info", fmt.Sprintf("error on GET protocols/nfs/export-policies: %s", err))
 	}
 
-	var dataONTAP ExportPolicyGetDataModelONTAP
+	var dataONTAP ExportpolicyResourceModel
 	if err := mapstructure.Decode(response, &dataONTAP); err != nil {
 		return nil, errorHandler.MakeAndReportError("error decoding export policy info", fmt.Sprintf("error on decode protocols/nfs/export-policies: %s, statusCode %d, response %#v", err, statusCode, response))
 	}
@@ -100,7 +101,7 @@ func GetNfsExportPolicyByName(errorHandler *utils.ErrorHandler, r restclient.Res
 }
 
 // GetExportPoliciesList to get export policies
-func GetExportPoliciesList(errorHandler *utils.ErrorHandler, r restclient.RestClient, filter *ExportPolicyGetDataFilterModel) ([]ExportpolicyResourceModel, error) {
+func GetExportPoliciesList(errorHandler *utils.ErrorHandler, r restclient.RestClient, filter *ExportPolicyGetDataFilterModel) ([]ExportPolicyGetDataModelONTAP, error) {
 	api := "protocols/nfs/export-policies"
 	query := r.NewQuery()
 	query.Fields([]string{"name", "id", "svm.name", "svm.uuid"})
@@ -119,9 +120,9 @@ func GetExportPoliciesList(errorHandler *utils.ErrorHandler, r restclient.RestCl
 		return nil, errorHandler.MakeAndReportError("error reading export policies info", fmt.Sprintf("error on GET %s: %s, statusCode %d", api, err, statusCode))
 	}
 
-	var dataONTAP []ExportpolicyResourceModel
+	var dataONTAP []ExportPolicyGetDataModelONTAP
 	for _, info := range response {
-		var record ExportpolicyResourceModel
+		var record ExportPolicyGetDataModelONTAP
 		if err := mapstructure.Decode(info, &record); err != nil {
 			return nil, errorHandler.MakeAndReportError(fmt.Sprintf("failed to decode response from GET %s", api),
 				fmt.Sprintf("error: %s, statusCode %d, info %#v", err, statusCode, info))

--- a/internal/interfaces/protocols_nfs_export_policy_test.go
+++ b/internal/interfaces/protocols_nfs_export_policy_test.go
@@ -13,12 +13,21 @@ import (
 	"github.com/netapp/terraform-provider-netapp-ontap/internal/utils"
 )
 
-// basic get data record
-var basicExportPolicyRecord = ExportPolicyGetDataModelONTAP{
+// basic get data records
+var oneBasicExportPolicyRecord = ExportPolicyGetDataModelONTAP{
 	Name:    "string",
 	Svm:     "string",
 	SvmUUID: "string",
 	ID:      123,
+}
+
+// basic get data record
+var basicExportPolicyRecord = ExportpolicyResourceModel{
+	Name: "string",
+	Svm: SvmDataModelONTAP{
+		Name: "string",
+		UUID: "string",
+	},
 }
 
 // bad record
@@ -91,7 +100,7 @@ func TestGetExportPolicy(t *testing.T) {
 	tests := []struct {
 		name      string
 		responses []restclient.MockResponse
-		want      *ExportPolicyGetDataModelONTAP
+		want      *ExportpolicyResourceModel
 		wantErr   bool
 	}{
 		{name: "test_no_records_1", responses: responses["test_no_records_1"], want: nil, wantErr: true},
@@ -147,7 +156,7 @@ func TestCreateExportPolicy(t *testing.T) {
 		name        string
 		responses   []restclient.MockResponse
 		requestbody ExportpolicyResourceBodyDataModelONTAP
-		want        *ExportPolicyGetDataModelONTAP
+		want        *ExportpolicyResourceModel
 		wantErr     bool
 	}{
 		{name: "test_create_basic_record_1", responses: responses["test_create_basic_record_1"], requestbody: basicExportPolicyBody, want: &basicExportPolicyRecord, wantErr: false},
@@ -255,7 +264,7 @@ func TestGetExportPoliciesList(t *testing.T) {
 	errorHandler := utils.NewErrorHandler(context.Background(), &diag.Diagnostics{})
 	badRecord := struct{ Name int }{123}
 	var recordInterface map[string]any
-	err := mapstructure.Decode(basicExportPolicyRecord, &recordInterface)
+	err := mapstructure.Decode(oneBasicExportPolicyRecord, &recordInterface)
 	if err != nil {
 		panic(err)
 	}
@@ -270,8 +279,8 @@ func TestGetExportPoliciesList(t *testing.T) {
 	twoRecordsResponse := restclient.RestResponse{NumRecords: 2, Records: []map[string]any{recordInterface, recordInterface}}
 	badRecordResponse := restclient.RestResponse{NumRecords: 1, Records: []map[string]any{badRecordInterface}}
 
-	var wantOneRecord = []ExportPolicyGetDataModelONTAP{basicExportPolicyRecord}
-	var wantTwoRecords = []ExportPolicyGetDataModelONTAP{basicExportPolicyRecord, basicExportPolicyRecord}
+	var wantOneRecord = []ExportPolicyGetDataModelONTAP{oneBasicExportPolicyRecord}
+	var wantTwoRecords = []ExportPolicyGetDataModelONTAP{oneBasicExportPolicyRecord, oneBasicExportPolicyRecord}
 
 	responses := map[string][]restclient.MockResponse{
 		"test_no_records_1": {

--- a/internal/interfaces/protocols_nfs_export_policy_test.go
+++ b/internal/interfaces/protocols_nfs_export_policy_test.go
@@ -24,38 +24,35 @@ var basicExportPolicyRecord = ExportPolicyGetDataModelONTAP{
 var badExportPolicyRecord = struct{ Name int }{123}
 
 // create export policy with basic request body
-var basicExportPolicyBody = ExportpolicyResourceModel{
+var basicExportPolicyBody = ExportpolicyResourceBodyDataModelONTAP{
 	Name: "string",
 	Svm: SvmDataModelONTAP{
 		Name: "string",
 		UUID: "string",
 	},
-	ID: 122880,
 }
 
 // create export policy with empty comment
-var badExportPolicyBody = ExportpolicyResourceModel{
+var badExportPolicyBody = ExportpolicyResourceBodyDataModelONTAP{
 	Name: "",
 }
 
 // update export policy name
-var renameExportPolicyBody = ExportpolicyResourceModel{
+var renameExportPolicyBody = ExportpolicyResourceBodyDataModelONTAP{
 	Name: "newname",
 	Svm: SvmDataModelONTAP{
 		Name: "string",
 		UUID: "string",
 	},
-	ID: 122880,
 }
 
 // update export policy with basic request body
-var updateExportPolicyErrorBody = ExportpolicyResourceModel{
+var updateExportPolicyErrorBody = ExportpolicyResourceBodyDataModelONTAP{
 	Name: "string",
 	Svm: SvmDataModelONTAP{
 		Name: "newsvm",
 		UUID: "string",
 	},
-	ID: 122880,
 }
 
 func TestGetExportPolicy(t *testing.T) {
@@ -148,7 +145,7 @@ func TestCreateExportPolicy(t *testing.T) {
 	tests := []struct {
 		name        string
 		responses   []restclient.MockResponse
-		requestbody ExportpolicyResourceModel
+		requestbody ExportpolicyResourceBodyDataModelONTAP
 		want        *ExportPolicyGetDataModelONTAP
 		wantErr     bool
 	}{
@@ -229,7 +226,7 @@ func TestUpdateExportPolicy(t *testing.T) {
 	tests := []struct {
 		name        string
 		responses   []restclient.MockResponse
-		requestbody ExportpolicyResourceModel
+		requestbody ExportpolicyResourceBodyDataModelONTAP
 		wantErr     bool
 	}{
 		{name: "test_update_rename_export_policy", responses: responses["test_update_rename_export_policy"], requestbody: renameExportPolicyBody, wantErr: false},
@@ -272,8 +269,8 @@ func TestGetExportPoliciesList(t *testing.T) {
 	twoRecordsResponse := restclient.RestResponse{NumRecords: 2, Records: []map[string]any{recordInterface, recordInterface}}
 	badRecordResponse := restclient.RestResponse{NumRecords: 1, Records: []map[string]any{badRecordInterface}}
 
-	var wantOneRecord = []ExportpolicyResourceModel{basicExportPolicyBody}
-	var wantTwoRecords = []ExportpolicyResourceModel{basicExportPolicyBody, basicExportPolicyBody}
+	var wantOneRecord = []ExportPolicyGetDataModelONTAP{basicExportPolicyRecord}
+	var wantTwoRecords = []ExportPolicyGetDataModelONTAP{basicExportPolicyRecord, basicExportPolicyRecord}
 
 	responses := map[string][]restclient.MockResponse{
 		"test_no_records_1": {
@@ -294,7 +291,7 @@ func TestGetExportPoliciesList(t *testing.T) {
 		name      string
 		responses []restclient.MockResponse
 		// args      args
-		want    []ExportpolicyResourceModel
+		want    []ExportPolicyGetDataModelONTAP
 		wantErr bool
 	}{
 		{name: "test_no_records_1", responses: responses["test_no_records_1"], want: nil, wantErr: false},

--- a/internal/interfaces/protocols_nfs_export_policy_test.go
+++ b/internal/interfaces/protocols_nfs_export_policy_test.go
@@ -17,7 +17,6 @@ import (
 var basicExportPolicyRecord = ExportPolicyGetDataModelONTAP{
 	Name: "string",
 	Svm:  "string",
-	ID:   122880,
 }
 
 // bad record

--- a/internal/interfaces/protocols_nfs_export_policy_test.go
+++ b/internal/interfaces/protocols_nfs_export_policy_test.go
@@ -15,8 +15,10 @@ import (
 
 // basic get data record
 var basicExportPolicyRecord = ExportPolicyGetDataModelONTAP{
-	Name: "string",
-	Svm:  "string",
+	Name:    "string",
+	Svm:     "string",
+	SvmUUID: "string",
+	ID:      123,
 }
 
 // bad record
@@ -253,7 +255,7 @@ func TestGetExportPoliciesList(t *testing.T) {
 	errorHandler := utils.NewErrorHandler(context.Background(), &diag.Diagnostics{})
 	badRecord := struct{ Name int }{123}
 	var recordInterface map[string]any
-	err := mapstructure.Decode(basicExportPolicyBody, &recordInterface)
+	err := mapstructure.Decode(basicExportPolicyRecord, &recordInterface)
 	if err != nil {
 		panic(err)
 	}

--- a/internal/interfaces/protocols_nfs_service.go
+++ b/internal/interfaces/protocols_nfs_service.go
@@ -2,6 +2,7 @@ package interfaces
 
 import (
 	"fmt"
+
 	"github.com/hashicorp/terraform-plugin-log/tflog"
 	"github.com/mitchellh/mapstructure"
 	"github.com/netapp/terraform-provider-netapp-ontap/internal/restclient"
@@ -10,6 +11,19 @@ import (
 
 // ProtocolsNfsServiceGetDataModelONTAP describes the GET record data model using go types for mapping.
 type ProtocolsNfsServiceGetDataModelONTAP struct {
+	Enabled          bool              `mapstructure:"enabled"`
+	Protocol         Protocol          `mapstructure:"protocol"`
+	Root             Root              `mapstructure:"root"`
+	Security         Security          `mapstructure:"security"`
+	ShowmountEnabled bool              `mapstructure:"showmount_enabled"`
+	Transport        Transport         `mapstructure:"transport"`
+	VstorageEnabled  bool              `mapstructure:"vstorage_enabled"`
+	Windows          Windows           `mapstructure:"windows"`
+	SVM              SvmDataModelONTAP `mapstructure:"svm"`
+}
+
+// ProtocolsNfsServiceResourceDataModelONTAP describes the GET record data model using go types for mapping.
+type ProtocolsNfsServiceResourceDataModelONTAP struct {
 	Enabled          bool              `mapstructure:"enabled"`
 	Protocol         Protocol          `mapstructure:"protocol"`
 	Root             Root              `mapstructure:"root"`
@@ -159,7 +173,7 @@ func GetProtocolsNfsServices(errorHandler *utils.ErrorHandler, r restclient.Rest
 }
 
 // CreateProtocolsNfsService Create a NFS Service
-func CreateProtocolsNfsService(errorHandler *utils.ErrorHandler, r restclient.RestClient, data ProtocolsNfsServiceGetDataModelONTAP, svnUUID string) (*ProtocolsNfsServiceGetDataModelONTAP, error) {
+func CreateProtocolsNfsService(errorHandler *utils.ErrorHandler, r restclient.RestClient, data ProtocolsNfsServiceResourceDataModelONTAP) (*ProtocolsNfsServiceGetDataModelONTAP, error) {
 	var body map[string]interface{}
 	if err := mapstructure.Decode(data, &body); err != nil {
 		return nil, errorHandler.MakeAndReportError("error encoding NFS Service body", fmt.Sprintf("error on encoding protocols/nfs/services body: %s, body: %#v", err, data))
@@ -188,7 +202,7 @@ func DeleteProtocolsNfsService(errorHandler *utils.ErrorHandler, r restclient.Re
 }
 
 // UpdateProtocolsNfsService Update a NFS service
-func UpdateProtocolsNfsService(errorHandler *utils.ErrorHandler, r restclient.RestClient, request ProtocolsNfsServiceGetDataModelONTAP, uuid string) error {
+func UpdateProtocolsNfsService(errorHandler *utils.ErrorHandler, r restclient.RestClient, request ProtocolsNfsServiceResourceDataModelONTAP, uuid string) error {
 	var body map[string]interface{}
 	if err := mapstructure.Decode(request, &body); err != nil {
 		return errorHandler.MakeAndReportError("error encoding NFS Services body", fmt.Sprintf("error on encoding NFS Services body: %s, body: %#v", err, request))

--- a/internal/interfaces/protocols_nfs_service_test.go
+++ b/internal/interfaces/protocols_nfs_service_test.go
@@ -201,8 +201,9 @@ func TestCreateProtocolsNfsService(t *testing.T) {
 		name      string
 		responses []restclient.MockResponse
 		// args      args
-		want    *ProtocolsNfsServiceGetDataModelONTAP
-		wantErr bool
+		requestBody ProtocolsNfsServiceResourceDataModelONTAP
+		want        *ProtocolsNfsServiceGetDataModelONTAP
+		wantErr     bool
 	}{
 		{name: "test_one_record_1", responses: responses["test_one_record_1"], want: &nfsServiceRecord, wantErr: false},
 		{name: "test_one_910_record_1", responses: responses["test_one_910_record_1"], want: &record910, wantErr: false},
@@ -215,7 +216,7 @@ func TestCreateProtocolsNfsService(t *testing.T) {
 			if err != nil {
 				panic(err)
 			}
-			got, err := CreateProtocolsNfsService(errorHandler, *r, nfsServiceRecord, "svmname")
+			got, err := CreateProtocolsNfsService(errorHandler, *r, tt.requestBody)
 			if err != nil {
 				fmt.Printf("err: %s\n", err)
 			}

--- a/internal/interfaces/storage_qtree.go
+++ b/internal/interfaces/storage_qtree.go
@@ -62,7 +62,6 @@ type StorageQtreeResourceBodyDataModelONTAP struct {
 	SVM             svm               `mapstructure:"svm"`
 	UnixPermissions int               `mapstructure:"unix_permissions,omitempty"`
 	Volume          qtreeVloume       `mapstructure:"volume"`
-	ID              int               `mapstructure:"id,omitempty"`
 	SecurityStyle   string            `mapstructure:"security_style,omitempty"`
 	ExportPolicy    qtreeExportPolicy `mapstructure:"export_policy,omitempty"`
 	User            qtreeUser         `mapstructure:"user,omitempty"`

--- a/internal/provider/protocols/protocols_nfs_export_policies_data_source.go
+++ b/internal/provider/protocols/protocols_nfs_export_policies_data_source.go
@@ -3,6 +3,7 @@ package protocols
 import (
 	"context"
 	"fmt"
+
 	"github.com/hashicorp/terraform-plugin-framework/datasource"
 	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
 	"github.com/hashicorp/terraform-plugin-framework/types"
@@ -160,8 +161,8 @@ func (d *ExportPoliciesDataSource) Read(ctx context.Context, req datasource.Read
 			CxProfileName: types.String(data.CxProfileName),
 			Name:          types.StringValue(record.Name),
 			ID:            types.Int64Value(int64(record.ID)),
-			SVMName:       types.StringValue(record.Svm.Name),
-			SVMUUID:       types.StringValue(record.Svm.UUID),
+			SVMName:       types.StringValue(record.Svm),
+			SVMUUID:       types.StringValue(record.SvmUUID),
 		}
 	}
 

--- a/internal/provider/protocols/protocols_nfs_export_policy_resource.go
+++ b/internal/provider/protocols/protocols_nfs_export_policy_resource.go
@@ -3,9 +3,10 @@ package protocols
 import (
 	"context"
 	"fmt"
-	"github.com/netapp/terraform-provider-netapp-ontap/internal/provider/connection"
 	"strconv"
 	"strings"
+
+	"github.com/netapp/terraform-provider-netapp-ontap/internal/provider/connection"
 
 	"github.com/hashicorp/terraform-plugin-framework/path"
 	"github.com/hashicorp/terraform-plugin-framework/resource"
@@ -104,7 +105,7 @@ func (r *ExportPolicyResource) Create(ctx context.Context, req resource.CreateRe
 	// Read Terraform plan data into the model
 	resp.Diagnostics.Append(req.Plan.Get(ctx, &data)...)
 
-	var request interfaces.ExportpolicyResourceModel
+	var request interfaces.ExportpolicyResourceBodyDataModelONTAP
 	errorHandler := utils.NewErrorHandler(ctx, &resp.Diagnostics)
 	if resp.Diagnostics.HasError() {
 		return
@@ -201,7 +202,7 @@ func (r *ExportPolicyResource) Update(ctx context.Context, req resource.UpdateRe
 		return
 	}
 
-	var request interfaces.ExportpolicyResourceModel
+	var request interfaces.ExportpolicyResourceBodyDataModelONTAP
 	request.Name = data.Name.ValueString()
 
 	err = interfaces.UpdateExportPolicy(errorHandler, *client, request, data.ID.ValueString())

--- a/internal/provider/protocols/protocols_nfs_service_resource.go
+++ b/internal/provider/protocols/protocols_nfs_service_resource.go
@@ -3,8 +3,9 @@ package protocols
 import (
 	"context"
 	"fmt"
-	"github.com/netapp/terraform-provider-netapp-ontap/internal/provider/connection"
 	"strings"
+
+	"github.com/netapp/terraform-provider-netapp-ontap/internal/provider/connection"
 
 	"github.com/hashicorp/terraform-plugin-framework/attr"
 	"github.com/hashicorp/terraform-plugin-framework/resource/schema/booldefault"
@@ -550,7 +551,7 @@ func (r *ProtocolsNfsServiceResource) Create(ctx context.Context, req resource.C
 	// Read Terraform plan data into the model
 	resp.Diagnostics.Append(req.Plan.Get(ctx, &data)...)
 
-	var body interfaces.ProtocolsNfsServiceGetDataModelONTAP
+	var body interfaces.ProtocolsNfsServiceResourceDataModelONTAP
 	errorHandler := utils.NewErrorHandler(ctx, &resp.Diagnostics)
 
 	if resp.Diagnostics.HasError() {
@@ -703,7 +704,7 @@ func (r *ProtocolsNfsServiceResource) Create(ctx context.Context, req resource.C
 		return
 	}
 
-	_, err = interfaces.CreateProtocolsNfsService(errorHandler, *client, body, svm.UUID)
+	_, err = interfaces.CreateProtocolsNfsService(errorHandler, *client, body)
 	if err != nil {
 		return
 	}
@@ -749,7 +750,7 @@ func (r *ProtocolsNfsServiceResource) Update(ctx context.Context, req resource.U
 		errorHandler.MakeAndReportError("No cluster found", "Cluster not found.")
 		return
 	}
-	var request interfaces.ProtocolsNfsServiceGetDataModelONTAP
+	var request interfaces.ProtocolsNfsServiceResourceDataModelONTAP
 	var errors []string
 	if !data.Enabled.IsNull() {
 		request.Enabled = data.Enabled.ValueBool()

--- a/internal/provider/storage/storage_qtrees_resource.go
+++ b/internal/provider/storage/storage_qtrees_resource.go
@@ -96,7 +96,6 @@ func (r *StorageQtreeResource) Schema(ctx context.Context, req resource.SchemaRe
 			"id": schema.Int64Attribute{
 				MarkdownDescription: "StorageQtree UUID",
 				Computed:            true,
-				Optional:            true,
 				PlanModifiers: []planmodifier.Int64{
 					int64planmodifier.UseStateForUnknown(),
 				},
@@ -312,9 +311,6 @@ func (r *StorageQtreeResource) Create(ctx context.Context, req resource.CreateRe
 	body.SVM.Name = data.SVMName.ValueString()
 	body.Volume.Name = data.Volume.ValueString()
 
-	if !data.ID.IsUnknown() {
-		body.ID = int(data.ID.ValueInt64())
-	}
 	if !data.SecurityStyle.IsUnknown() {
 		body.SecurityStyle = data.SecurityStyle.ValueString()
 	}

--- a/internal/provider/storage/storage_qtrees_resource_test.go
+++ b/internal/provider/storage/storage_qtrees_resource_test.go
@@ -35,7 +35,7 @@ func TestAccStorageQtreesResource(t *testing.T) {
 			{
 				ResourceName:  "netapp-ontap_storage_qtrees.example",
 				ImportState:   true,
-				ImportStateId: fmt.Sprintf("%s,%s,%s,%s", "acc_import", "temp_root", "temp", "cluster5"),
+				ImportStateId: fmt.Sprintf("%s,%s,%s,%s", "acc_import", "terraform_root", "terraform", "cluster5"),
 				Check: resource.ComposeTestCheckFunc(
 					resource.TestCheckResourceAttr("netapp-ontap_storage_qtrees.example", "name", "accFlexcache"),
 				),

--- a/internal/provider/storage/storage_qtrees_resource_test.go
+++ b/internal/provider/storage/storage_qtrees_resource_test.go
@@ -35,7 +35,7 @@ func TestAccStorageQtreesResource(t *testing.T) {
 			{
 				ResourceName:  "netapp-ontap_storage_qtrees.example",
 				ImportState:   true,
-				ImportStateId: fmt.Sprintf("%s,%s,%s,%s", "acc_import", "terraform_root", "terraform", "cluster5"),
+				ImportStateId: fmt.Sprintf("%s,%s,%s,%s", "acc_import", "acc_test_root", "acc_test", "cluster2"),
 				Check: resource.ComposeTestCheckFunc(
 					resource.TestCheckResourceAttr("netapp-ontap_storage_qtrees.example", "name", "accFlexcache"),
 				),

--- a/internal/provider/storage/storage_qtrees_resource_test.go
+++ b/internal/provider/storage/storage_qtrees_resource_test.go
@@ -35,9 +35,9 @@ func TestAccStorageQtreesResource(t *testing.T) {
 			{
 				ResourceName:  "netapp-ontap_storage_qtrees.example",
 				ImportState:   true,
-				ImportStateId: fmt.Sprintf("%s,%s,%s,%s", "acc_import", "acc_test_root", "acc_test", "cluster2"),
+				ImportStateId: fmt.Sprintf("%s,%s,%s,%s", "acc_import", "acc_test_root", "acc_test", "cluster5"),
 				Check: resource.ComposeTestCheckFunc(
-					resource.TestCheckResourceAttr("netapp-ontap_storage_qtrees.example", "name", "accFlexcache"),
+					resource.TestCheckResourceAttr("netapp-ontap_storage_qtrees.example", "name", "acc_import"),
 				),
 			},
 		},

